test/unit_tests/braket/experimental/autoqasm/test_transpiler.py

```diff
@@ -71,50 +68,74 @@ def bell(q0: int, q1: int):
-        bell(q0, q1)
+    expected_partial = """OPENQASM 3.0;
+input int[32] q1;
+qubit[2] __qubits__;
+h __qubits__[1];
+cnot __qubits__[1], __qubits__[q1];"""
 
-    expected = """OPENQASM 3.0;
-qubit[4] __qubits__;
+    expected_no_arg_partial = """OPENQASM 3.0;
+qubit[2] __qubits__;
 h __qubits__[1];
 cnot __qubits__[1], __qubits__[3];"""
 
-    bell_partial = aq.main(functools.partial(bell, 1))
-    assert bell_partial(3).to_ir() == expected
+    bell_partial = aq.main(num_qubits=2)(functools.partial(bell, 1))
+    assert bell_partial.to_ir() == expected_partial
 
-    bell_decorated_partial = functools.partial(bell_decorated, 1)
-    assert bell_decorated_partial(3).to_ir() == expected
-
-    bell_noarg_partial = functools.partial(bell_decorated, 1, 3)
-    assert bell_noarg_partial().to_ir() == expected
+    bell_noarg_partial = aq.main(num_qubits=2)(functools.partial(bell, 1, 3))
+    assert bell_noarg_partial.to_ir() == expected_no_arg_partial
 
 
 def test_classmethod() -> None:
-    """Tests aq.main decorator application to a classmethod."""
+    """Tests aq.main application to a classmethod."""
+    # Note - this only works with the function call syntax, since with a decorator,
+    # the class isn't initialized fully at transpilation time.
 
     class MyClass:
         @classmethod
-        def bell(self, q0: int, q1: int):
+        def bell(cls, q0: int, q1: int):
             h(q0)
             cnot(q0, q1)
 
-        @classmethod
-        @aq.main
-        def bell_decorated(self, q0: int, q1: int):
-            self.bell(q0, q1)
+    expected = """OPENQASM 3.0;
+input int[32] q0;
+input int[32] q1;
+qubit[2] __qubits__;
+h __qubits__[q0];
+cnot __qubits__[q0], __qubits__[q1];"""
+
+    assert aq.main(num_qubits=2)(MyClass.bell).to_ir() == expected
+    assert aq.main(num_qubits=2)(MyClass().bell).to_ir() == expected
+
+
+def test_method() -> None:
+    """Tests aq.main decorator application to a classmethod."""
+
+    # todo: see if this functionality should work
+    pytest.xfail("can't access `self` at function definition time")
+
+    class MyClass:
+        def __init__(self):
+            self.class_param = 2
+
+        def rx(self, q0: int):
+            rx(target=q0, angle=self.class_param)
+
+        @aq.main(num_qubits=1)
+        def rx_decorated(self, q0: int):
+            self.rx(q0)
 
     expected = """OPENQASM 3.0;
-qubit[4] __qubits__;
-h __qubits__[1];
-cnot __qubits__[1], __qubits__[3];"""
+input int[32] q0;
+qubit[1] __qubits__;
+rx __qubits__[q0];"""
 
-    assert aq.main(MyClass.bell)(1, 3).to_ir() == expected
-    assert MyClass.bell_decorated(1, 3).to_ir() == expected
+    assert aq.main(num_qubits=2)(MyClass.rx).to_ir() == expected
+    assert MyClass.rx_decorated.to_ir() == expected
 
     a = MyClass()
-    assert aq.main(a.bell)(1, 3).to_ir() == expected
-    assert a.bell_decorated(1, 3).to_ir() == expected
+    assert aq.main(num_qubits=2)(a.rx).to_ir() == expected
```

Comment:
Shouldn't `rx` be called with `q0`?

Reply:
This line passes the function `a.rx` to the decorator; the function isn't actually called at all. You can see from the expected ir that `q0` is captured from the signature as an input. Regardless, I will delete this test as mentioned above